### PR TITLE
Add draft of spawn and world rendering implementation

### PR DESCRIPTION
**Pull request by EduardoLR10**

### CODE REVIEW

File: client/src/game/physics.zig [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+const GameState = @import("../game/state.zig");
+const std = @import("std");
+
+const heightAxis: rl.Vector3 = .{
+    .x = 0,
+    .y = 1,
+    .z = 0,
+};
+
+pub const Character = struct {
```
Comment (z-silver) [inline client/src/game/physics.zig:12]: This is just a namespace, not a struct meant for instantiation. Don't use an uppercase name.

File: client/src/menu/character.zig [hunk 2]
```diff
@@ -132,47 +142,36 @@ pub fn selection(gameState: *GameState) !void {
...
+    switch (maybe_characters) {
+        .ok => |erlang_characters| {
 
-                    const teapotImage = try assets.image("teapot.png");
+            // todo: discover how to make this work
+            // const teapotembed = @embedfile("../assets/teapot.png");
+            // const teapotloaded = rl.loadimagefrommemory(".png", teapotembed, teapotembed.len);
 
-                    var characters = std.ArrayList(messages.Character).init(gameState.allocator);
+            const teapotimage = try assets.image("teapot.png");
```
Comment (z-silver) [inline client/src/menu/character.zig:159]: These two words are just glued together. They are painful to read.

File: client/src/server_messages.zig [hunk 3]
```diff
@@ -1,124 +1,141 @@
...
-    user_registry,
-    user_login,
-    character_list,
-    debug,
-};
 
-pub const Erlang_Response = union(enum) {
-    ok: void,
-    @"error": [:0]const u8,
+fn createAnonymousStruct(comptime T: type, comptime keys: []const [:0]const u8) type {
```
Comment (z-silver) [inline client/src/server_messages.zig:5]: We're not interoping with C. We don't need the slice to be null-terminated.

File: justfile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 client:
-    cd client && zig build -fsys=raylib --search-prefix ${ERLANG_INTERFACE_PATH} --search-prefix ${RAYLIB_PATH} run -- \"$@\"
+    cd client && zig build --search-prefix ${ERLANG_INTERFACE_PATH} --search-prefix ${RAYLIB_PATH} run -- \"$@\"
```
Comment (z-silver) [inline justfile:2]: We should probably mirror this change to `client-build` as well.

File: src/server.erl [hunk 5]
```diff
@@ -30,40 +30,76 @@ database_connect() ->
...
+			 StreamPid = spawn(?MODULE, stream_user_data, [NewState]),
+			 NewState = State#{stream_pid := StreamPid},
+			 %% TimerOutPid = spawn(?MODULE, timeout_user, [NewState]),
+			 handle_user(NewState)
+	    end;
+	{update_character, Character_Map} ->
+	    io:format("Character will be updated"),
+	    character:updateTemp(Character_Map, Connection),
+	    UserPid ! ok;
+	Something ->
```
Comment (z-silver) [inline src/server.erl:71]: The catch all clause will prevent the `retrieve_character` clause from ever being reached.

File: client/src/server_messages.zig (hunk 3 above)
Comment (EduardoLR10) [reply client/src/server_messages.zig:5]: The entire refactoring of null-terminated strings will be addressed later on.

**Review by z-silver — APPROVED** [review]

**Pull request merged**